### PR TITLE
reduce empty graph noise

### LINKED ISSUE #9966: New ScriptCanvas assets saved from the ScriptCanvas editor fail in the asset pipeline
**Description**
There is an error and a warning when saving a newly created scriptcanvas asset from the ScriptCanvas editor.

Example logs from the screenshot below:
```(System) - Failed to store dynamic module information, increase size of AZ::Debug::SymbolStorageDynamicallyLoadedModules.m_modules to maintain stack tracer functionality!
(System) - Failed to store dynamic module information, increase size of AZ::Debug::SymbolStorageDynamicallyLoadedModules.m_modules to maintain stack tracer functionality!
(System) - Failed to store dynamic module information, increase size of AZ::Debug::SymbolStorageDynamicallyLoadedModules.m_modules to maintain stack tracer functionality!
[Error] (ScriptCanvas) - Script Canvas parsing failed
(ScriptCanvas) - Ignoring source file modification notification (possibly external), as a it was recently saved by the editor: ScriptCanvas/zNewScript.scriptcanvas
[Warning] (AssetManager) - GetAsset called for asset which does not exist in asset catalog and cannot be loaded.  Asset may be missing, not processed or moved.  AssetId: {40EC881D-571D-53A4-BF93-F625FB523B42}:163310ae
```

**Steps to reproduce**
1. Open the ScripCanvas editor.
2. Select File > New Script.
3. Select File > Save.
4. Name the script and save it.
5. ---> There is an error about failed parsing and a warning about the asset being missing, not processed or moved.

**Expected behavior**
The script will save without any issues.

**Actual behavior**
The script appears to save and processes properly in the Asset Processor but messages in the log indicate something might be wrong.

**Screenshots/Video**
Screenshot showing the errors and the asset's missing icon in the Asset Browser, indicating that something went wrong:
errorWhenSavingNewScriptCanvasAsset

### PR DESCRIPTION
**Pull request by carlitosan**
Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

## What does this PR do?

Reduces the noise generated by empty graphs

Closes #9966

## How was this PR tested?

Repeatedly saving empty graphs.

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

**Review by AMZN-daimini — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp [hunk 1]
```diff
@@ -984,7 +985,7 @@ namespace AZ::Data
             }
             else
             {
-                AZ_Warning("AssetManager", false, "GetAsset called for asset which does not exist in asset catalog and cannot be loaded.  Asset may be missing, not processed or moved.  AssetId: %s",
+                AZLOG_WARN("GetAsset called for asset which does not exist in asset catalogand cannot be loaded.Asset may be missing, not processed or moved.AssetId: % s",
```
Comment (carlitosan) [inline Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: @lawsonamzn looking for your opinion on this one

Comment (lawsonamzn) [reply Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: Why the removal of spaces and compaction?

Note:  This is probably the wrong place to make changes to reduce noise.  This AssetManager serves the entire engine, the graphics, audio, animation, levels, everything, and this function is used to warn that an asset is asked for that doesn't exist (or doesn't exist yet).  Removing it from view is going to be unhelpful when the real solution is generally to either NOT call GetAsset unless you're sure such an asset exists, or, to use the Asset PRocessor Api (like CompileAssetSync) to ensure that the Asset Processor generates the asset if its in the queue, before you call GetAsset.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Graph.cpp [hunk 2]
```diff
@@ -316,7 +318,12 @@ namespace ScriptCanvas
 
             if (validationResults.HasResults())
             {
-                AZ_Error("ScriptCanvas", result.IsModelValid(), "Script Canvas parsing failed");
+                // reduce the noise of reporting against empty graphs
+                if (!(validationResults.ErrorCount() == 1
```
Comment (onecent1101) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Graph.cpp:322]: if we don't log error for empty graph, do we handle empty graph in other ways? (like some notifications to tell user that graph is empty) if not, then have an EmptyGraph specific validation result doesn't make sense. Or we should log it in debug build?

**Review by onecent1101 — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp (hunk 1 above)
Comment (carlitosan) [reply Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: In this particular situation, a system receives a SourceFileChanged notification for a file it knows will get processed by the AP. So then it connects to the asset bus on the source file uuid, and calls queue load to receive the OnAssetReady (or Error) calls back. If it can recieve those notifications without trying to load the asset, I will do that, but that's how I thought this had to happen.

Comment (carlitosan) [reply Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: removal of whitespace and compaction was just a copy paste error.

Comment (lawsonamzn) [reply Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: When SourceFileChanged happens, it means the source file has changed but not necessarily that a product is ready.  It can take time to process it.  A different message is available if you want to listen for assets changing in the cache.

In fact, trying to load an _asset_ when a _source_ changes is a recipe for problems because you'll almost always get Source File Changed some time before the asset exists (especially on first compile).  

Note that this error/warning message happens when it cannot even find the asset in the catalog - ie, it does not exist (yet).   Its usually an error to request something that does not exist.

In the engine, Asset<T> is generally mean to reference an in memory loaded asset.  If you want to work with the asset system API on the file level (instead of on the asset level), then you can hook into the catalog bus broadcasts, such as AssetCatalogEvents OnCatalogAssetChanged/OnCatalogAssetAdded, which does not require you to subscribe to an address and will tell you about every asset change.  These happen at the file and catalog level, wheras OnAssetready / OnAssetReloaded is specifically for assets that are already loaded or being loaded.

Note that AssetIds are just SourceFileID + SubId typles, so if you know a source id, you can listen for its assets to be done by keeping track of what sources you're looking out for and listening ton OnCatalogAssetChanged/OnCatalogAssetAdded.  

So basically, if you just want to know about changes in asset data on the file system, use the AssetCatalogEvents bus.  If you have _already_ loaded an asset (so you already have an Asset<T>) you can use the OnAssetReady / OnAssetReloaded to track changes to that specific instance of that specific asset.    The full sequence of events for a not-currently-loaded asset is something like

OnSourceFileChanged
...some time passes ..
 OnCatalogAssetChanged/OnCatalogAssetAdded (when the products become available and have been compiled)
  ... some more time passes..
OnAssetReloaded/OnAssetReady (if and only if there is a currently loaded instance of that asset<T> in memory and also only if the system that owns that asset type has opted into default behavior for reloading).

Comment (carlitosan) [reply Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp:988]: Cool, I'll work with the catalog events.

File: Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp [hunk 3]
```diff
@@ -1231,7 +1217,6 @@ namespace ScriptCanvasEditor
 
     void MainWindow::RemoveScriptCanvasAsset(const SourceHandle& assetId)
     {
-        AZ_TracePrintf("ScriptCanvas", "RemoveScriptCanvasAsset : %s\n", assetId.ToString().c_str());
```
[the remark this answers is not in the document]
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp:1234]: There are better status reporting in the builder system to reflect this.

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Graph.cpp (hunk 2 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Graph.cpp:322]: We track the error, still, and it is used in the builder to NOT build an empty asset, but we will update the reporting of it.

**Review by lsemp3d — APPROVED** [review]
